src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs

```diff
@@ -3464,778 +3595,407 @@ protected override bool CopyToFast(KeyValuePair<string, StringValues>[] array, i
+                    {
+                        flag = 0x4L;
+                        values = ref _headers._Date;
+                    }
+                    else if ((firstTerm4 == 0x4d4f5246u))
+                    {
+                        flag = 0x4000000L;
+                        values = ref _headers._From;
+                    }
+                    break;
+                case 5:
+                    var firstTerm5 = (Unsafe.ReadUnaligned<uint>(ref nameStart) & 0xdfdfdfdfu);
+                    if ((firstTerm5 == 0x4f4c4c41u) && ((Unsafe.AddByteOffset(ref nameStart, (IntPtr)4) & 0xdfu) == 0x57u))
+                    {
+                        flag = 0x400L;
+                        values = ref _headers._Allow;
+                    }
+                    else if ((firstTerm5 == 0x474e4152u) && ((Unsafe.AddByteOffset(ref nameStart, (IntPtr)4) & 0xdfu) == 0x45u))
+                    {
+                        flag = 0x1000000000L;
+                        values = ref _headers._Range;
+                    }
+                    break;
+                case 6:
+                    var firstTerm6 = (Unsafe.ReadUnaligned<uint>(ref nameStart) & 0xdfdfdfdfu);
+                    if ((firstTerm6 == 0x45434341u) && ((Unsafe.ReadUnaligned<ushort>(ref Unsafe.AddByteOffset(ref nameStart, (IntPtr)(2 * sizeof(ushort)))) & 0xdfdfu) == 0x5450u))
+                    {
+                        flag = 0x80000L;
+                        values = ref _headers._Accept;
+                    }
+                    else if ((firstTerm6 == 0x4b4f4f43u) && ((Unsafe.ReadUnaligned<ushort>(ref Unsafe.AddByteOffset(ref nameStart, (IntPtr)(2 * sizeof(ushort)))) & 0xdfdfu) == 0x4549u))
+                    {
+                        flag = 0x1000000L;
+                        values = ref _headers._Cookie;
+                    }
+                    else if ((firstTerm6 == 0x45505845u) && ((Unsafe.ReadUnaligned<ushort>(ref Unsafe.AddByteOffset(ref nameStart, (IntPtr)(2 * sizeof(ushort)))) & 0xdfdfu) == 0x5443u))
+                    {
+                        flag = 0x2000000L;
+                        values = ref _headers._Expect;
+                    }
+                    else if ((firstTerm6 == 0x4749524fu) && ((Unsafe.ReadUnaligned<ushort>(ref Unsafe.AddByteOffset(ref nameStart, (IntPtr)(2 * sizeof(ushort)))) & 0xdfdfu) == 0x4e49u))
+                    {
+                        flag = 0x40000000000L;
+                        values = ref _headers._Origin;
+                    }
+                    else if ((firstTerm6 == 0x47415250u) && ((Unsafe.ReadUnaligned<ushort>(ref Unsafe.AddByteOffset(ref nameStart, (IntPtr)(2 * sizeof(ushort)))) & 0xdfdfu) == 0x414du))
+                    {
+                        flag = 0x10L;
+                        values = ref _headers._Pragma;
+                    }
+                    break;
+                case 7:
+                    var firstTerm7 = (Unsafe.ReadUnaligned<uint>(ref nameStart) & 0xdfdfdfdfu);
```

Comment:
Potential optimization: Don't be afraid to perform overlapping reads. You can read the first 4 bytes as a DWORD and the last 4 bytes as a DWORD. Sure, there's going to be one overlapping byte, but the fact that fewer overall operations are necessary should still result in a net perf boost.

Reply:
Will do in follow up